### PR TITLE
[WIP] feat!: bump workspace dependencies with release

### LINKED ISSUE #8: cargo-smart-release doesn't update `[workspace.dependencies]` version
First off, this crate is awesome. It is *so close* to being a does-what-you-want publish tool for cargo crates.

For [this project](https://github.com/candlecorp/wick) though, we [set our version numbers in the root workspace](https://github.com/candlecorp/wick/blob/main/Cargo.toml#L90-L137) and `cargo-smart-release` doesn't seem to update things there. When the publish step comes, cargo fails because versions don't match.

If organizing workspaces like this is bad practice, I can change it. That's one major problem with large Rust projects and workspaces. There is no "do it this way" guide that actually works from start to publish.

I tried to reorganize the project to align with git-oxide. I moved the binary to the root and combined the package & workspace Cargo.toml, but it wasn't enough.

If this seems like a bug in `cargo-smart-release` vs how we structured our project, I'll gladly submit a PR if you can guide me to the part of the codebase I should start in.

Thanks again!

### PR DESCRIPTION
**Pull request by brooksmtownsend**
Fixes #8 

This PR adds support for bumping workspace dependencies just like how the current smart-releaser will bump dependent crates in response to a new release. In order to implement this, I just slotted some logic in with the existing logic to find dependent crates to also look for a `workspace.dependencies` section in the Cargo.toml.

This works... mostly. It actually works perfectly for dependencies that the main workspace uses, but it does not work when a dependency is specified at the workspace level but not used by the main workspace.

For example, I was taking a look at this for wasmCloud. If I use this to bump the crate [wasmcloud-core](https://github.com/wasmCloud/wasmCloud/blob/main/Cargo.toml#L96), which our primary package depends on, I get the expected result (wasmcloud-core bumps from 0.5 to 0.6, workspace dependency updates. However, if I use this to bump [wash-lib](https://github.com/wasmCloud/wasmCloud/blob/main/Cargo.toml#L270), this is specified as a workspace dependency but wasmCloud itself doesn't use it, so no update. I wanted to open up this PR as a draft WIP while I work through this just to let you know I'm working on it 😄 

Some trace logs from a run (will remove extraneous trace logs when I mark this as ready of course):
```console
[TRACE] Updating wasmcloud
[TRACE] Crates that may change wasmcloud-core, opentelemetry-nats, provider-archive, wasmcloud-runtime, wasmcloud-tracing, wasmcloud-host, wasmcloud-provider-sdk, wasmcloud-provider-blobstor
e-fs, wasmcloud-provider-blobstore-s3, wasmcloud-provider-http-client, wasmcloud-provider-http-server, wasmcloud-provider-keyvalue-redis, wasmcloud-provider-keyvalue-vault, wasmcloud-provide
r-lattice-controller, wasmcloud-provider-messaging-kafka, wasmcloud-provider-messaging-nats, wasmcloud, wash-cli, wash-lib
[TRACE] Found workspace dependencies
[TRACE] Pending 'wasmcloud' manifest workspace-dependencies update: 'wasmcloud-core = "^0.6.0"' (from  "0.5")
[TRACE] Pending 'wasmcloud' conservative manifest workspace-dependencies update: 'provider-archive = "^0.10.1"' (from  "0.10")
[TRACE] Pending 'wasmcloud' manifest workspace-dependencies update: 'wasmcloud-tracing = "^0.4.0"' (from  "0.3")
[TRACE] Pending 'wasmcloud' manifest workspace-dependencies update: 'wasmcloud-provider-sdk = "^0.5.0"' (from  "0.4")
```

### CODE REVIEW

File: src/command/release/manifest.rs [hunk 1]
```diff
@@ -67,6 +67,15 @@
                 None,
             )?),
         };
+        log::trace!("Updating {}", package.name);
+        log::trace!(
+            "Crates that may change {}",
+            crates_with_version_change
+                .iter()
+                .map(|(p, _)| format!("{}", p.name))
```
[the remark this answers is not in the document]
Comment (brooksmtownsend) [reply src/command/release/manifest.rs]: Don't be so mean, it's just debug code! 😡

**Comment by brooksmtownsend** [comment]
I think test is failing because I added extra output.

Thinking out loud, I think I need to consider a workspace root as something that may always change in response to a version bump, and that'll work well to catch my edge case

**Comment by brooksmtownsend** [comment]
@Byron what kind of tests could we add to add confidence in not breaking `gitoxide`? I'd be happy to guinea pig test this release, though I can only speak to a few use cases 😕

**Comment by Byron** [comment]
I don't think given this requirement, there isn't anything less but to publish `gitoxide` itself, and such a publish can come in many different shapes. Impossible to model these. On the bright side, that's the only requirement, so if I manage to create one release with a version from this PR, I'd assume it's not breaking anything.

But like a said, a simple Journey test should probably be added to generally show how it works with workspace dependencies.

**Comment by brooksmtownsend** [comment]
Just a note my latest commit isn't quite ready for code review, but it does work! There's some edge cases I have called out in the TODO, and more to test, but I'm planning on trying it out to release a few crates this week (likely via manual inspection)

**Comment by Byron** [comment]
By the way, I tried to publish with this branch but it failed with an issue around invalid manifests being written. `cargo metadata` failed due to that. I am sorry to not have the specific output, I forgot to save it while it was still available in the terminals history.

**Comment by brooksmtownsend** [comment]
Thank you for letting me know @Byron , I think an issue here (with my implementation) is still that

- [ ] smart-release with workspace dependencies still edits manifest without `--execute`

If that wasn't the issue though, let me know what the error is and I can try to track it down 🙂

**Comment by brooksmtownsend** [comment]
Yeah I bet if you have any pre-check step then the above bug will cause an issue, e.g. run without `--execute` to double check, then run with `--execute`
